### PR TITLE
Fixes #12555: switched to session-based freeipa api

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by dmitri-d** [comment]
[test]

File: modules/realm/freeipa.rb [hunk 1]
```diff
@@ -39,17 +40,23 @@ def initialize
...
-        gssapi = GSSAPI::Simple.new(@ipa_server.host, "HTTP")
-        token = gssapi.init_context
+        @gssapi = GSSAPI::Simple.new(@ipa_server.host, "HTTP")
+        token = @gssapi.init_context
+
+        login = Net::HTTP.new(@ipa_server.host, 443)
+        login.use_ssl = true
+        login.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+        request = Net::HTTP::Post.new("/ipa/session/login_kerberos", 'Authorization'=>"Negotiate #{strict_encode64(token)}", 'Referer' => @ipa_server.to_s)
```
Comment (domcleal) [inline modules/realm/freeipa.rb:50]: Is this available in all versions, or just with the most recent update/change?

Comment (dmitri-d) [inline modules/realm/freeipa.rb:50]: Looks like support for sessions for xml api was introduced in freeipa version 3.0. That login end-point first appeared in version 2.2, I think.

**Comment by domcleal** [comment]
@stbenjam did you want to look this over at all?

**Comment by domcleal** [comment]
Merged as 07a6dae, thanks @witlessbird.

**Pull request closed without merging**